Prototype guidance

davidhunter08 opened the pull request:
Updating the prototyping guidance to help designers get setup with the NHS App prototype.

## First draft 

![image](https://github.com/user-attachments/assets/023a1a4c-26c9-4911-b97f-830e9a1d732b)

Trying to reduce the duplication of content with the prototype kit as much as possible by linking out to guidance.

Tosin-Balogun commented:
I'll have a look at this PR tomorrow @davidhunter08

Tosin-Balogun commented:
## Pull request review

I just realised I have not downloaded the microsite instance to inspect the branch on my machine, so I just used the screenshot on the PR here.

This is an improvement on the previous version for sure, somewhere in the middle between an expert and novice user.

### Some things to consider (ignore if not seen as helpful)
- Assuming the novice user who is less familiar with code has come here, would they know what to look for in terms of downloading the app prototype kit on GitHub?

- Where it says "NHS App design system and NHS App prototype", can it just say "NHS app prototype and design system"

- Where it says "You must follow the NHS prototype kit setup guide first", can it use a less forceful tone and say something that implies that's what they need to install first like "To get going, you will need to install the kit by following the NHS prototype kit setup guide first", rewrite this for the intended style

- Another thing worth considering is the link to the NHS kit takes the user away from this get started guide. Which means by the time the novice user has gone there to complete the task, they prob have forgotten where they started from. I don't know if a 'opens in a new window' would be useful here.

- The last copy where it says "You can copy the code from the HTML or Nunjucks tabs below any examples" is very confusing. I had to read it 3 times and it's still confusing. If the idea is to introduce the concept of the user copying code from the tabs, perhaps focus on just this one instance and when next they see it on the guides, they would recognise it. That copy is trying to do too much here when the user is meant to be focused on getting started

Tosin-Balogun reviewed (changes requested):
## Pull request review

I just realised I have not downloaded the microsite instance to inspect the branch on my machine, so I just used the screenshot on the PR here.

This is an improvement on the previous version for sure, somewhere in the middle between an expert and novice user.

### Some things to consider (ignore if not seen as helpful)
- Assuming the novice user who is less familiar with code has come here, would they know what to look for in terms of downloading the app prototype kit on GitHub?

- Where it says "NHS App design system and NHS App prototype", can it just say "NHS app prototype and design system"

- Where it says "You must follow the NHS prototype kit setup guide first", can it use a less forceful tone and say something that implies that's what they need to install first like "To get going, you will need to install the kit by following the NHS prototype kit setup guide first", rewrite this for the intended style

- Another thing worth considering is the link to the NHS kit takes the user away from this get started guide. Which means by the time the novice user has gone there to complete the task, they prob have forgotten where they started from. I don't know if a 'opens in a new window' would be useful here.

- The last copy where it says "You can copy the code from the HTML or Nunjucks tabs below any examples" is very confusing, I had to read it 3 times and it's still confusing. If the idea is to introduce the concept of the user copying code from the tabs, perhaps focus on just this one instance and when next they see it on the guides, they would recognise it. That copy is trying to do too much here when the user is meant to be focused on getting started

davidhunter08 commented:
Thanks for looking @Tosin-Balogun 

> Assuming the novice user who is less familiar with code has come here, would they know what to look for in terms of downloading the app prototype kit on GitHub?

Assumption is that they would have done the first part and followed the NHS prototype kit guidance, so then the 'downloading' bit would be more familiar. But there might be a better way to explain this?

> Where it says "NHS App design system and NHS App prototype", can it just say "NHS app prototype and design system"

Makes sense to me, I'll change it then see what @Graham-Pembrey thinks.

> Where it says "You must follow the NHS prototype kit setup guide first", can it use a less forceful tone and say something that implies that's what they need to install first like "To get going, you will need to install the kit by following the NHS prototype kit setup guide first", rewrite this for the intended style

Good point. @Graham-Pembrey any thoughts on this?

> Another thing worth considering is the link to the NHS kit takes the user away from this get started guide. Which means by the time the novice user has gone there to complete the task, they prob have forgotten where they started from. I don't know if a 'opens in a new window' would be useful here.

Definitely something to consider. 

> The last copy where it says "You can copy the code from the HTML or Nunjucks tabs below any examples" is very confusing, I had to read it 3 times and it's still confusing. If the idea is to introduce the concept of the user copying code from the tabs, perhaps focus on just this one instance and when next they see it on the guides, they would recognise it. That copy is trying to do too much here when the user is meant to be focused on getting started

Ok, I'll rethink about this copy 🤔.

anna-rigo commented:
> ## Pull request review
> 
> I just realised I have not downloaded the microsite instance to inspect the branch on my machine, so I just used the screenshot on the PR here.
> 
> This is an improvement on the previous version for sure, somewhere in the middle between an expert and novice user.
> ### Some things to consider (ignore if not seen as helpful)
> 
>     * Assuming the novice user who is less familiar with code has come here, would they know what to look for in terms of downloading the app prototype kit on GitHub?
> 
>     * Where it says "NHS App design system and NHS App prototype", can it just say "NHS app prototype and design system"
> 
>     * Where it says "You must follow the NHS prototype kit setup guide first", can it use a less forceful tone and say something that implies that's what they need to install first like "To get going, you will need to install the kit by following the NHS prototype kit setup guide first", rewrite this for the intended style
> 
>     * Another thing worth considering is the link to the NHS kit takes the user away from this get started guide. Which means by the time the novice user has gone there to complete the task, they prob have forgotten where they started from. I don't know if a 'opens in a new window' would be useful here.
> 
>     * The last copy where it says "You can copy the code from the HTML or Nunjucks tabs below any examples" is very confusing, I had to read it 3 times and it's still confusing. If the idea is to introduce the concept of the user copying code from the tabs, perhaps focus on just this one instance and when next they see it on the guides, they would recognise it. That copy is trying to do too much here when the user is meant to be focused on getting started

In regards to Tosin's last point, could it help to maybe highlight the HTML and Nunjucks options in the example (e.g., red box)? Of course if there is a way to be consistent with NHS design. 

I don't want to be nitpicking at all, this is just a suggestion. Would it help to be consistent with images used in line with our App design system so having the example image underneath the line "You can use..." and then the text below the image?

Graham-Pembrey commented:
### Intro 

Good points about the intro sentences. Just a suggestion here for tweaking it slightly and perhaps mentioning the value of prototyping?

> This guide explains how to use the NHS App prototype. You can use it alongside the NHS App design system to make interactive prototypes that look like real NHS App pages. 
> 
> Anyone can use the prototype to: 
> 
> - try out ideas
> - get feedback from users or people you work with


### Before you start

Tosin made a good observation about tone. I think his suggested text works well. Agreed that it might be best to open the external links a new window. I know the Service Manual generally advises caution about doing that, but it gives some exceptions. It feels like it makes sense here? https://service-manual.nhs.uk/content/links 


### Download and Guides

On the points about novice users, as one myself, I'm wondering if it might be better to put the 'Guides' section first and then 'Download'? We could perhaps advise people to get familiar with the NHS prototype kit first, and _then_ download the NHS App prototype on GitHub? I feel that's probably what I need to do before using the app kit.

### Example

I agree that Anna's idea is worth considering. Another option could be changing the line above the example to explain a bit more. For example:

> On the NHS and NHS App design systems, you'll find examples of components like the one below. To use any of these components in your prototype, you can select **HTML** or **Nunjucks**, then copy and paste the code.

davidhunter08 commented:
## Second draft

![image](https://github.com/user-attachments/assets/9917a14e-df3f-42c3-a689-f1c9ba860b5f)

Updates based on feedback:
- updated intro with @Graham-Pembrey's suggestion
- softened the 'before you start' copy based on @Tosin-Balogun's feedback
- moved 'Guides' before 'Download'
- updated the 'Components' copy with @Graham-Pembrey's suggestion (also thanks @anna-rigo)

Notes:
- the second sentence I added in the 'Get started' section feels a bit clunky - cc @Graham-Pembrey

Tosin-Balogun commented on docs/get-started/prototyping.md:
```diff
@@ -5,28 +5,27 @@ tags:
+## Using components
 
-You can copy the code from the HTML or Nunjucks tabs below any examples:
+On the NHS and NHS App design systems, you'll find examples of components like the one below. To use any of these components in your prototype, you can select HTML or Nunjucks, then copy and paste the code.
```
"then copy and paste the code into your prototype"

davidhunter08 replied:
Change made 👍

Tosin-Balogun reviewed (changes requested):
A lot of great improvements from the previous version.

The guide section stands out, I have added comment to that section. I think that is all, prob good to go

Tosin-Balogun commented on docs/get-started/prototyping.md:
```diff
@@ -5,28 +5,27 @@ tags:
-Download or clone a copy of the [NHS App prototype on GitHub](https://github.com/nhsuk/nhsapp-prototype).
+You first need to follow the [NHS prototype kit setup guide](https://prototype-kit.service-manual.nhs.uk/install) to make sure you have the tools and software required to run the prototype. This is because the NHS App prototype is a copy of the NHS prototype kit, with added NHS App code and templates.
 
 ## Guides
```
The 'get started' page while not a full tutorial, is a form of tutorial itself. It is the first exposure the user would have to doing a self-help installation and use of the kit.

Part of the reason I made the suggestion is also because I have recently used Apple & Google's guidance as well and it follows the pattern I mentioned prior. It started with little steps and wins which made me feel like I was slowly building towards something.

Essentially, you are slowly exposing the user to concepts such as installing, trying it out (rewarding them with a successful trial) and then sending them to the wider resource. The guide being in the middle disrupts this momentum because the guide is broad and what the user need here is something more focused, little steps and little wins that helps their momentum and builds their confidence.

It makes sense for them to copy the code snippet on this page and add it to their kit, so they see the instant result of how it works, they get a joyful feeling and their confidence grows. Now, then they can go to the wider guides to see what else is there.

davidhunter08 replied:
Ok, makes sense, so:

- Learn what the prototype kit is
- Get the tools/software required so I can use it (by following the NHS prototype install guide)
- Start using the NHS App prototype (by downloading a copy)
- Try out a sample component
- Understand more about how to use it (by following the NHS prototype kit guides)
- Get support if needed

davidhunter08 commented:
## Third draft 

![image](https://github.com/user-attachments/assets/9a18a96f-4dcd-4814-959f-60485595b307)

davidhunter08 commented:
> A lot of great improvements from the previous version.
> 
> The guide section stands out, I have added comment to that section. I think that is all, prob good to go

I made changes based on your feedback @Tosin-Balogun 👀

Tosin-Balogun reviewed (approved):
Thanks Dave – Looks super good 🚀 🚀 🚀
